src/robot/errors.py

```diff
@@ -139,14 +139,14 @@ def can_continue(self, context, templated=False):
         if self.syntax or self.exit or self.skip or self.test_timeout:
             return False
         if templated:
-            return True
+            return context.continue_on_failure(default=True)
         if self.keyword_timeout:
             if context.in_teardown:
                 self.keyword_timeout = False
             return False
-        if context.in_teardown or context.continue_on_failure:
-            return True
-        return self.continue_on_failure
+        if context.in_teardown:
+            return context.continue_on_failure(default=True)
+        return self.continue_on_failure or context.continue_on_failure(default=False)
```

Comment:
Adding `default=False` argument to `continue_on_failure` looks like a good idea.

I got an idea that the three last lines could be simplified to this:
```python
return self.continue_on_failure or context.continue_on_failure(default=context.in_teardown)
```
Passing `context.in_teardown` to `context.continue_on_failure` is a bit odd, though, as `context` obviously has that information already. Perhaps we should just use
```python
return self.continue_on_failure or context.continue_on_failure()
```
here and then at the end of the `continue_on_failure` method change
```python
return default
```
to
```python
return default or self.in_teardown
```

Reply:
Somehow I feel it is cleaner to leave the in_teardown check in `can_continue` so the `continue_on_failure` is purely about checking the tags and no other side-effects..  
I think `context.continue_on_failure(default=context.in_teardown)` looks odd only at first glance.. changing it to `context.continue_on_failure(default=context.in_teardown is True)` would be more explicit, but I reckon this would void the optimization (not sure how big such an optimization would be)?

Reply:
there is actually a behavioral difference: In the current case, context.in_teardown overrides continuable keywords, so in the current implementation the user-keyword stops, ignoring continuable keyword. 

```
stop-on-failure with run-kw-and-continue failure in keyword in Teardown
    [Documentation]     FAIL    Teardown failed:\nkw10a
    [Teardown]   run-kw-and-continue failure in user keyword with stop tag
    No Operation

*** Keywords ***
run-kw-and-continue failure in user keyword with stop tag
    [Tags]   robot:stop-on-failure
    Run Keyword And Continue On Failure    Fail    kw10a
    Log    This is not executed
    Fail    kw10b
```

Combining the two statements into one line 
```
return self.continue_on_failure or context.continue_on_failure(default=context.in_teardown)
```
would continue the execution in the keyword.

Reply:
Ok, understood.. not sure how common continuable keywords are, but if clearly documented people should be able to handle this.
changes implemented through 4f3fe30ab